records/state/state.go

```diff
@@ -123,12 +131,22 @@ var sources = map[string]func(*Task) []string{
 func hostIPs(t *Task) []string { return []string{t.SlaveIP} }
 
 // networkInfoIPs returns IP addresses from a given Task's
-// []Status.ContainerStatus.[]NetworkInfos.IPAddress
+// []Status.ContainerStatus.[]NetworkInfos.[]IPAddresses.IPAddress
 func networkInfoIPs(t *Task) []string {
 	return statusIPs(t.Statuses, func(s *Status) []string {
 		ips := make([]string, len(s.ContainerStatus.NetworkInfos))
-		for i, netinfo := range s.ContainerStatus.NetworkInfos {
-			ips[i] = netinfo.IPAddress
+		for _, netinfo := range s.ContainerStatus.NetworkInfos {
+			if len(netinfo.IPAddresses) > 0 {
```

Comment:
I think that without the if/else, we'd pick the 0.25 syntax IP if both
versions were present for some reason - this way we pick the 0.26 version
in that case.

On Wed, Nov 18, 2015 at 8:16 AM James DeFelice <notifications@github.com>
wrote:

> In records/state/state.go
> <https://github.com/mesosphere/mesos-dns/pull/355#discussion_r45220278>:
>
> >  func networkInfoIPs(t *Task) []string {
> >  	return statusIPs(t.Statuses, func(s *Status) []string {
> >  		ips := make([]string, len(s.ContainerStatus.
>
> NetworkInfos))
> > -		for i, netinfo := range s.ContainerStatus.NetworkInfos {
> > -			ips[i] = netinfo.IPAddress
> > +		for _, netinfo := range s.ContainerStatus.NetworkInfos {
> > +			if len(netinfo.IPAddresses) > 0 {
>
> the if len(..) > 0 {} else {} is redundant. simplify to:
>
> for _, ipAddress := range netinfo.IPAddresses {
> ...
> }if netinfo.IPAddress != "" {
> ...
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/mesos-dns/pull/355/files#r45220278>.
>